runtime/libia2/ia2.c

```diff
@@ -344,19 +344,25 @@ int protect_pages(struct dl_phdr_info *info, size_t size, void *data) {
 
   struct PhdrSearchArgs *search_args = (struct PhdrSearchArgs *)data;
 
-  size_t cur_pkey = ia2_get_tag();
+  size_t cur_pkey = search_args->pkey;//ia2_get_tag();
```

Comment:
I'd either remove the comment, or expand it to explain why it used to be `ia2_get_tag` and you're changing it.